### PR TITLE
MOSIP-25617,MOSIP-25716 :Updated db scripts

### PR DESCRIPTION
**Pull request by balaji-alluru**

### CODE REVIEW

File: partner/partner-management-service/src/main/java/io/mosip/pms/oidc/client/dto/ClientDetailCreateRequest.java [hunk 1]
```diff
@@ -27,7 +27,7 @@ public class ClientDetailCreateRequest {
     @NotNull
     @NotBlank
```
Comment (ase-101) [inline partner/partner-management-service/src/main/java/io/mosip/pms/oidc/client/dto/ClientDetailCreateRequest.java:28]: @notNull is not required when using @notBlank

File: partner/partner-management-service/src/main/java/io/mosip/pms/oidc/client/dto/ClientDetailCreateRequest.java [hunk 2]
```diff
@@ -27,7 +27,7 @@ public class ClientDetailCreateRequest {
...
     @Size(min = 1, max = 128)
```
Comment (ase-101) [inline partner/partner-management-service/src/main/java/io/mosip/pms/oidc/client/dto/ClientDetailCreateRequest.java:29]: is size check required here?

File: partner/partner-management-service/src/main/java/io/mosip/pms/oidc/client/service/impl/ClientManagementServiceImpl.java [hunk 3]
```diff
@@ -107,30 +112,41 @@ public ClientDetailResponse createOIDCClient(ClientDetailCreateRequest createReq
...
+				.findByPartnerIdAndPolicyId(createRequest.getAuthPartnerId(), policyFromDb.get().getId());
 		if (policyMappingReqFromDb.isEmpty()) {
 			LOGGER.error("createOIDCClient::Policy and partner mapping not exists for policy {} and partner {}",
-					createRequest.getPolicyName(), createRequest.getAuthPartnerId());
+					createRequest.getPolicyId(), createRequest.getAuthPartnerId());
 			throw new PartnerServiceException(ErrorCode.PARTNER_POLICY_MAPPING_NOT_EXISTS.getErrorCode(),
 					ErrorCode.PARTNER_POLICY_MAPPING_NOT_EXISTS.getErrorMessage());
 		}
 
 		if (!policyMappingReqFromDb.get(0).getStatusCode().equalsIgnoreCase("approved")) {
```
Comment (ase-101) [inline partner/partner-management-service/src/main/java/io/mosip/pms/oidc/client/service/impl/ClientManagementServiceImpl.java:146]: why not take "approved" from constants.

File: partner/partner-management-service/src/main/java/io/mosip/pms/oidc/client/dto/ClientDetailCreateRequest.java (hunk 1 above)
Comment (balaji-alluru) [reply partner/partner-management-service/src/main/java/io/mosip/pms/oidc/client/dto/ClientDetailCreateRequest.java:26]: Updated the same anusha

File: partner/partner-management-service/src/main/java/io/mosip/pms/oidc/client/service/impl/ClientManagementServiceImpl.java [hunk 4]
```diff
@@ -107,30 +112,41 @@ public ClientDetailResponse createOIDCClient(ClientDetailCreateRequest createReq
...
+			throw new PartnerServiceException(ErrorCode.INVALID_PARTNERID.getErrorCode(), String
+					.format(ErrorCode.INVALID_PARTNERID.getErrorMessage(), createRequest.getAuthPartnerId()));
+		}
+		if(!partner.get().getPartnerTypeCode().equalsIgnoreCase(AUTH_PARTNER_TYPE)) {
+			LOGGER.error("createOIDCClient::{} cannot create OIDC Client", partner.get().getPartnerTypeCode());
+			throw new PartnerServiceException(ErrorCode.INVALID_PARTNER_TYPE.getErrorCode(), String
+					.format(ErrorCode.INVALID_PARTNER_TYPE.getErrorMessage(), partner.get().getPartnerTypeCode()));
+		}
+		Optional<AuthPolicy> policyFromDb = authPolicyRepository.findById(createRequest.getPolicyId());
+				if (policyFromDb == null) {
```
[the remark this answers is not in the document]
Comment (balaji-alluru) [reply partner/partner-management-service/src/main/java/io/mosip/pms/oidc/client/service/impl/ClientManagementServiceImpl.java]: Updated the same anusha

File: partner/partner-management-service/src/main/java/io/mosip/pms/oidc/client/dto/ClientDetailCreateRequest.java (hunk 2 above)
Comment (balaji-alluru) [reply partner/partner-management-service/src/main/java/io/mosip/pms/oidc/client/dto/ClientDetailCreateRequest.java:27]: Removed NotNull annotations

**Review by ase-101 — APPROVED** [review]

**Pull request merged**